cc_types.c

```diff
@@ -290,6 +292,68 @@ void create_struct(void)
 	i->members = last;
 }
 
+void create_enum(void)
+{
+	struct type* head = calloc(1, sizeof(struct type));
+	require(NULL != head, "Exhausted memory while creating an enum\n");
+
+	head->name = global_token->s;
+	head->type = head;
+	head->indirect = NULL;
+	head->next = global_types;
+
+	head->size = register_size; /* We treat enums as always being ints. */
+	head->is_signed = TRUE;
+
+	global_types = head;
+
+	global_token = global_token->next;
+	require_match("ERROR in create_enum\n Missing {\n", "{");
+
```

Comment:
I think you are missing `require(NULL != global_token, "Incomplete enum definition at end of file\n");`

Reply:
I had that before but as far as I can tell `require_match` also checks for `NULL`. I suppose the error message would be better with an explicit `require`.

Reply:
I think `require_match` checks for null before that token you require. But then the last line of `require_match` advances token with `global_token = global_token->next;`so before you can use `global_token->next` you again have to check for it to be non NULL, otherwise you can crash M2-Planet.